actors/runtime/src/runtime/fvm.rs

```diff
@@ -263,28 +261,44 @@ where
         value: TokenAmount,
     ) -> Result<RawBytes, ActorError> {
         if self.in_transaction {
-            return Err(actor_error!(SysErrIllegalActor; "runtime.send() is not allowed"));
+            return Err(actor_error!(user_assertion_failed; "runtime.send() is not allowed"));
         }
-        // TODO: distinguish between "syscall" errors and actor exit codes.
         match fvm::send::send(&to, method, params, value) {
             Ok(ret) => {
                 if ret.exit_code.is_success() {
                     Ok(ret.return_data)
                 } else {
-                    Err(ActorError::from(ret.exit_code))
+                    // The returned code can't be simply propagated as it may be a system exit code.
+                    // TODO: improve propagation once we return a RuntimeError.
```

Comment:
Reference an issue for this, perhals #144.

Reply:
fixed

Reply:
This may work for now, but this isn't the right fix. The caller _should_ be able to determine which exit code the called actor failed with.

Can we propagate the error as something other than an `ActorError` so we don't automatically propagate it up the chain?

Reply:
I've changed this to propagate non-system exit codes, and convert _system_ exit codes to either "unspecified" or "assertion failed" (depending on the exit code).